codec-http/src/main/java/io/netty/handler/codec/http/websocketx/extensions/compression/UncompressedWebSocketFrame.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */package io.netty.handler.codec.http.websocketx.extensions.compression;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.handler.codec.http.websocketx.BinaryWebSocketFrame;
+import io.netty.handler.codec.http.websocketx.TextWebSocketFrame;
+import io.netty.handler.codec.http.websocketx.WebSocketFrame;
+import io.netty.handler.codec.http.websocketx.extensions.WebSocketExtension;
+
+/**
+ * A frame wrapper for selective websocket compression.
+ * Wrap your frame with this class if you want it uncompressed.
+ */
+public class UncompressedWebSocketFrame extends WebSocketFrame {
+
+    private boolean isBinaryFrame = true;
+
+    /**
+     * Wrap the websocket frame to skip compression.
+     *
+     * @param frame the frame to skip compression.
+     */
+    public UncompressedWebSocketFrame(WebSocketFrame frame) {
+        this(frame.isFinalFragment(), frame.rsv(), frame.content());
+
+        if (! ((frame instanceof TextWebSocketFrame) || (frame instanceof BinaryWebSocketFrame))) {
+            throw new IllegalArgumentException("Text frame or binary frame expected.");
+        }
+
+        if (frame instanceof TextWebSocketFrame) {
+            isBinaryFrame = false;
+        }
+    }
+
+    private UncompressedWebSocketFrame(boolean isFinalFragment, int rsv, ByteBuf content) {
+        super(isFinalFragment, rsv, content);
+        if ((rsv & WebSocketExtension.RSV1) > 0) {
+            throw new IllegalStateException("This frame is already compressed.");
+        }
+    }
+
+    @Override
+    public UncompressedWebSocketFrame replace(ByteBuf content) {
+        UncompressedWebSocketFrame frame = new UncompressedWebSocketFrame(isFinalFragment(), rsv(), content);
+        frame.isBinaryFrame = isBinaryFrame;
+
+        return frame;
+    }
+
+    public WebSocketFrame unwrap() {
```

Comment:
@normanmaurer do you have any suggestion on naming this? I call it `unwrap` because this class acts as a wrapper of WebSocketFrame, so here we extract the actual frame for further usage.

Also this function is only used in PerMessageDeflateEncoder, do you think it's a good idea to make it package private?